### PR TITLE
elk: 6.2.4 -> 6.3.2

### PR DESCRIPTION
**Pull request by basvandijk**
~~**This is WIP. Please don't merge yet!**~~

###### Motivation for this change

https://www.elastic.co/guide/en/elasticsearch/reference/6.3/release-notes-6.3.0.html

Also the private X-Pack is now opened up and moved into the ELK repositories. See: https://www.elastic.co/products/x-pack/open

###### Things done

* The ELK stack is upgraded to 6.3.0.

* `elasticsearch6`, `logstash6` and `kibana6` now come with X-Pack which is
  a suite of additional features. These are however licensed under the unfree
  "Elastic License".

* Fortunately they also provide OSS versions which are now packaged
  under: `elasticsearch6-oss`, `logstash6-oss` and `kibana6-oss`.
  Note that the naming of the attributes is consistent with upstream.

* The test `nix-build nixos/tests/elk.nix -A ELK-6` will test the OSS
  version by default. You can also run the test on the unfree ELK using:

  `NIXPKGS_ALLOW_UNFREE=1 nix-build nixos/tests/elk.nix -A ELK-6 --arg enableUnfree true`

  ~~That still fails however. Working on it...~~

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by basvandijk** [comment]
@GrahamcOfBorg test elk.ELK-6

File: pkgs/development/tools/misc/kibana/6.x.nix [hunk 1]
```diff
@@ -1,17 +1,45 @@
-{ stdenv, makeWrapper, fetchurl, elk6Version, nodejs, coreutils, which }:
+{ elk6Version
+, enableUnfree ? true
```
Comment (LnL7) [inline pkgs/development/tools/misc/kibana/6.x.nix]: Why would we want `pkgs.kibana` to be unfree by default? We don't enable allowUnfree by default so this package wouldn't be available.

File: pkgs/servers/search/elasticsearch/6.x.nix [hunk 2]
```diff
@@ -1,14 +1,26 @@
-{ stdenv, fetchurl, elk6Version, makeWrapper, jre_headless, utillinux }:
+{ elk6Version
+, enableUnfree ? true
```
Comment (LnL7) [inline pkgs/servers/search/elasticsearch/6.x.nix]: Same as above.

**Review by LnL7 — COMMENTED** [review]
I noticed elasticsearch/kibana disappeared from hydra in 18.09, I don't think we want that unless there's a good very reason to do so.

File: pkgs/development/tools/misc/kibana/6.x.nix (hunk 1 above)
Comment (basvandijk) [reply pkgs/development/tools/misc/kibana/6.x.nix]: Also see: https://github.com/NixOS/nixpkgs/commit/3f889f30bee92c3f2d5eaf53e56aef68f61efc4a

**Pull request merged**